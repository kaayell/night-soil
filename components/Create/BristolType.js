import React from 'react'
import { Text, TouchableOpacity, View, } from 'react-native'
import { Button } from 'react-native-elements'
import { BLUE, OFF_WHITE } from '../StyleGuide/colors'

const labelStyle = {fontSize: 10, color: OFF_WHITE}

export const BristolType = (props) => {
  const backgroundColor = props.selectedBristolType === props.type
    ? OFF_WHITE
    : 'transparent'
  const fontColor = props.selectedBristolType === props.type ? BLUE : OFF_WHITE
  return (
    <Button
      title={`${props.type}: ${props.label1} ${props.label2 || ''}`}
      buttonStyle={{
        backgroundColor: backgroundColor,
        width: 90,
        height: 45,
        borderColor: OFF_WHITE,
        borderWidth: .5,
        borderRadius: 5,
      }}
      textStyle={
        {
          ...labelStyle,
          color: fontColor,
        }
      }
      onPress={() => {
        console.log(props.type)
        props.setPoopRating(props.type)
      }}/>)
}