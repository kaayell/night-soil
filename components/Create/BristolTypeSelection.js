import React from 'react'
import { Text, View } from 'react-native'
import { FormLabel } from 'react-native-elements'
import { POPPINS } from '../StyleGuide/fonts'
import { RadioButton, RadioGroup } from 'react-native-flexi-radio-button'

export const BristolTypeSelection = (props) => {
  return (
    <View>
      <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>Bristol Type</FormLabel>
      <RadioGroup onSelect={(index, value) => props.setBristolType(value)}>
        <RadioButton value={'1'}><Text>1 (Separate hard lumps)</Text></RadioButton>
        <RadioButton value={'2'}><Text>2 (Lumpy and sausage like)</Text></RadioButton>
        <RadioButton value={'3'}><Text>3 (Cracked sausage shape)</Text></RadioButton>
        <RadioButton value={'4'}><Text>4 (Smooth sausage)</Text></RadioButton>
        <RadioButton value={'5'}><Text>5 (Soft blobs with clear edges)</Text></RadioButton>
        <RadioButton value={'6'}><Text>6 (Mushy with ragged edges)</Text></RadioButton>
        <RadioButton value={'7'}><Text>7 (Liquid)</Text></RadioButton>
      </RadioGroup>
    </View>
  )
}