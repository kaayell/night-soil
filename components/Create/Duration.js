import React from 'react'
import { POPPINS } from '../StyleGuide/fonts'
import { OFF_WHITE } from '../StyleGuide/colors'
import { View } from 'react-native'
import { FormInput, FormLabel } from 'react-native-elements'

export const Duration = ({onTextFieldChange}) => {
  return <View style={{flex: 1}}>
    <FormLabel labelStyle={{color: 'white', fontSize: 16, paddingBottom: 10}} fontFamily={POPPINS}>
      Duration
    </FormLabel>
    <View style={{flex: 1, flexDirection: 'row', width: '100%'}}>
      <FormInput
        onChangeText={(text) => onTextFieldChange('durationMinutes', text)}
        containerStyle={{
          backgroundColor: 'transparent',
          borderColor: OFF_WHITE,
          borderWidth: .5,
          borderRadius: 5,
          height: '50%',
          width: '20%',
          marginRight: -10
        }}
        inputStyle={{
          color: OFF_WHITE,
          fontSize: 16,
          paddingLeft: 10
        }}
        underlineColorAndroid='rgba(0,0,0,0)'
      />
      <FormLabel labelStyle={{color: 'white', fontSize: 18}} fontFamily={POPPINS}>min</FormLabel>
    </View>
  </View>
}

export default Duration