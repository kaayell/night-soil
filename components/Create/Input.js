import React from 'react'
import { View } from 'react-native'
import { FormInput, FormLabel } from 'react-native-elements'
import { POPPINS } from '../StyleGuide/fonts'

export const Input = (props) => {
  return (
    <View>
      <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>{props.labelText}</FormLabel>
      <FormInput onChangeText={(text) => props.onTextFieldChange(props.stateField, text)}/>
    </View>
  )
}

