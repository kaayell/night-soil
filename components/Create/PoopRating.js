import React from 'react'
import {FormLabel} from 'react-native-elements'
import {View} from 'react-native'
import {POPPINS} from '../StyleGuide/fonts'
import StarRating from "react-native-star-rating";

export const PoopRating = ({selected, onRatingChange}) => {
  const labelText = [
    'Kill Me',
    'Shouldn\'t have had spicy wings',
    'Average',
    'Quick and easy',
    'Did I even poop?']
  return <View style={{flex: 1}}>
    <FormLabel
      labelStyle={{color: 'white', fontSize: 20, textAlign: 'center', paddingBottom: 10}}
      fontFamily={POPPINS}
    >
      {labelText[selected - 1]}
    </FormLabel>
    <StarRating
      maxStars={5}
      disabled={false}
      rating={selected}
      fullStarColor={"FFCC00"}
      selectedStar={onRatingChange}
      containerStyle={{flex: 1, justifyContent: 'center', alignItems: 'center'}}
    />
  </View>
}

export default PoopRating