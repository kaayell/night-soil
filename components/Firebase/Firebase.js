import * as firebase from 'firebase'
import { ANDROID_API_KEY, config, IOS_API_KEY } from './secrets'

export default class Firebase {
  static init () {
    firebase.initializeApp(config)
  }

  static getAuth() {
    return firebase.auth()
  }

  static getUserDetailsRef() {
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/user/details/${uid}`)
  }

  static savePoop(poopDetails) {
    let currentUser = firebase.auth().currentUser
    firebase.database().ref(`/poop/${currentUser.uid}`).push().set(poopDetails)
  }

  static getPoopsRef(){
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/poop/${uid}`)
  }

  static async loginWithGoogle () {
    const response = await Expo.Google.logInAsync({
      androidClientId: ANDROID_API_KEY,
      iosClientId: IOS_API_KEY,
      scopes: ['profile', 'email'],
    })

    if (response.type === 'success') {
      const credential = firebase.auth.GoogleAuthProvider.credential(response.idToken, response.accessToken)
      firebase.auth().signInWithCredential(credential)
    }
  }
}