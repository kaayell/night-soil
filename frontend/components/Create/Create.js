import React, { Component } from 'react'
import * as apiClient from '../../api/apiClient'
import { connect } from 'react-redux'
import { clearTime } from '../Timer/timer-actions'
import { Picker, View } from 'react-native'
import BaseView from '../BaseView/BaseView'
import { FormInput, FormLabel, Icon } from 'react-native-elements'
import { POPPINS, POPPINS_MEDIUM } from '../StyleGuide/fonts'
import style from '../StyleGuide/icon-styles'
import { BLUE, OFF_WHITE } from '../StyleGuide/colors'

export class Create extends Component {
  static navigationOptions = ({navigation}) => ({
    title: 'Create',
    headerStyle: {
      backgroundColor: BLUE,
      borderBottomWidth: 0
    },
    headerTitleStyle: {
      color: OFF_WHITE,
      fontFamily: POPPINS_MEDIUM
    },
    headerLeft: null,
    headerRight: <Icon name={'clear'} iconStyle={style.icon} color={'white'}
                       onPress={() => navigation.goBack()} underlayColor={BLUE}/>
  })

  constructor (props) {
    super(props)

    this.handleSubmit = this.handleSubmit.bind(this)
    this.onDropDownChange = this.onDropDownChange.bind(this)
    this.onDateChange = this.onDateChange.bind(this)

    this.state = {
      bristolType: null,
      durationInMinutes: this.props.poopTime,
      comments: null,
      dateTimeInMilliseconds: null,

      errorTexts: {
        bristolTypeErrorText: '',
        durationInMinutesErrorText: '',
        commentsErrorText: '',
        dateTimeInMillisecondsErrorText: ''
      }
    }
  }

  handleSubmit () {
    let errorTexts = {}
    Object.entries(this.state).forEach(([key, value]) => {
      if (value === null) {
        errorTexts[`${key}ErrorText`] = 'This field is required'
      }
    })

    if (Object.keys(errorTexts).length !== 0) {
      this.setState({errorTexts: {...this.state.errorTexts, ...errorTexts}})
      return
    }

    apiClient.createLog({...this.state, ...{humanId: this.props.humanId}})
    this.props.clearTime()
    this.props.setActivePage('home')
  }

  onDropDownChange (event, index, value) {
    this.setState({
      bristolType: value,
      errorTexts: {...this.state.errorTexts, ...{bristolTypeErrorText: ''}}
    })
  }

  onDateChange (event, date) {
    this.setState({
      dateTimeInMilliseconds: date.getTime(),
      errorTexts: {...this.state.errorTexts, ...{dateTimeInMillisecondsErrorText: ''}}
    })
  }

  onTextFieldChange (field, value) {
    const errorObj = {}
    errorObj[`${field}ErrorText`] = ''
    const obj = {}
    obj[field] = value === '' ? null : value
    obj['errorTexts'] = {...this.state.errorTexts, ...errorObj}
    this.setState(obj)
  }

  render () {
    return (
      <View style={{backgroundColor: BLUE, height: '100%'}}>
        <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>Bristol Type</FormLabel>
        <Picker>
          <Picker.Item label="1 (Separate hard lumps)" value="1"/>
          <Picker.Item label="2 (Lumpy and sausage like)" value="2"/>
          <Picker.Item label="3 (Cracked sausage shape)" value="3"/>
          <Picker.Item label="4 (Smooth sausage)" value="4"/>
          <Picker.Item label="5 (Soft blobs with clear edges)" value="5"/>
          <Picker.Item label="6 (Mushy with ragged edges)" value="6"/>
          <Picker.Item label="7 (Liquid)" value="7"/>
        </Picker>
        <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>Duration</FormLabel>
        <FormInput onChangeText={this.onTextFieldChange.bind(this, 'durationInMinutes')}/>
        <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>Comments</FormLabel>
        <FormInput onChangeText={this.onTextFieldChange.bind(this, 'comments')}/>
        <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>Date</FormLabel>
      </View>
    )
  }
}

const mapStateToProps = state => {
  return {
    humanId: state.humanInfo.id,
    poopTime: state.poopTime
  }
}

export default connect(mapStateToProps, {clearTime})(Create)