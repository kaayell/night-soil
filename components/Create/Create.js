import React, { Component } from 'react'
import { View } from 'react-native'
import { Button, FormLabel, Icon } from 'react-native-elements'
import { POPPINS, POPPINS_MEDIUM } from '../StyleGuide/fonts'
import style from '../StyleGuide/styles'
import { BLUE, OFF_WHITE } from '../StyleGuide/colors'
import DatePicker from 'react-native-datepicker'
import Firebase from '../Firebase/Firebase'
import moment from 'moment'
import { BristolTypeSelection } from './BristolTypeSelection'
import { Input } from './Input'

export default class Create extends Component {
  static navigationOptions = ({navigation}) => ({
    title: 'Create',
    headerStyle: {
      backgroundColor: BLUE,
      borderBottomWidth: 0,
    },
    headerTitleStyle: {
      color: OFF_WHITE,
      fontFamily: POPPINS_MEDIUM,
    },
    headerLeft: null,
    headerRight: <Icon name={'clear'} iconStyle={style.icon} color={'white'}
                       onPress={() => navigation.goBack()}
                       underlayColor={BLUE}/>,
  })

  constructor (props) {
    super(props)

    this.handleSubmit = this.handleSubmit.bind(this)
    this.setBristolType = this.setBristolType.bind(this)
    this.onTextFieldChange = this.onTextFieldChange.bind(this)

    this.state = {
      bristolType: null,
      duration: null,
      comments: null,
      date: moment().format('MM-DD-YYYY'),
    }
  }

  handleSubmit () {
    Firebase.savePoop(this.state)
    this.props.navigation.goBack()
  }

  setBristolType (bristolType) {
    this.setState({bristolType})
  }

  onTextFieldChange (field, value) {
    const obj = {}
    obj[field] = value === '' ? null : value
    this.setState(obj)
  }

  render () {
    return (
      <View style={{backgroundColor: BLUE, height: '100%'}}>
        <BristolTypeSelection
          selectedBristolType={this.state.bristolType}
          setBristolType={this.setBristolType}/>
        <Input labelText={'Duration'} stateField={'duration'}
               onTextFieldChange={this.onTextFieldChange}/>
        <Input labelText={'Comments'} stateField={'comments'}
               onTextFieldChange={this.onTextFieldChange}/>
        <FormLabel labelStyle={{color: 'white', fontSize: 16}}
                   fontFamily={POPPINS}>Date</FormLabel>
        <DatePicker showIcon={false} style={{width: 'auto'}}
                    date={this.state.date}
                    mode="date" format="MM-DD-YYYY" confirmBtnText="Confirm"
                    cancelBtnText="Cancel"
                    customStyles={{
                      placeholderText: {color: 'white', fontSize: 16},
                      dateText: {color: 'white', fontSize: 16},
                      dateInput: {borderColor: 'transparent'},
                    }}
                    onDateChange={(date) => {this.setState({date: date})}}
        />
        <Button
          title='ADD'
          onPress={() => this.handleSubmit()}
          textStyle={{color: OFF_WHITE, fontWeight: '700'}}
          buttonStyle={style.button}
          containerViewStyle={{flex: 1, alignItems: 'center'}}
        />
      </View>
    )
  }
}