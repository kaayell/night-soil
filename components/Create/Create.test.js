import {shallow} from "enzyme";
import Create from "./Create";
import React from "react";
import {FormInput} from "react-native-elements";
import DatePicker from "react-native-datepicker";
import Firebase from "../Firebase/Firebase";

describe('Create', () => {
  let wrapper, fakeCloseModal

  beforeEach(() => {
    Firebase.getUserDetailsRef = () => {
      return {
        once: () => Promise.resolve({val: () => {
          return {salary: "20"}
        }})
      }
    }
    fakeCloseModal = jest.fn()

    wrapper = shallow(<Create closeModal={fakeCloseModal}/>)
  })

  it('should render a whole lot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('duration', () => {
    it('should set state on change', () => {
      wrapper.find(FormInput).at(0).props().onChangeText("3")
      expect(wrapper.instance().state.durationMinutes).toEqual("3")
    });
  });

  describe('date', () => {
    it('should set state on change', () => {
      wrapper.find(DatePicker).props().onDateChange("03-20-2018")
      expect(wrapper.instance().state.date).toEqual("03-20-2018")
    });
  });

  describe('work toggle', () => {
    it('should toggle state', () => {
      wrapper.find('Button').at(0).props().onPress()
      expect(wrapper.instance().state.atWork).toEqual(true)
    });
  });

  describe('poop rating', () => {
    it('should pass through function', () => {
      expect(wrapper.find('PoopRating').props().onRatingChange).toEqual(wrapper.instance().setPoopRating)
    });
  });

  describe('onSubmit', () => {
    it('should set errors if duration is not added', () => {
      wrapper.find('Button').at(1).props().onPress()
      expect(wrapper.instance().state.errors).toEqual(true)
    });

    it('should close the modal', () => {
      wrapper.setState({durationMinutes: "2"})
      wrapper.find('Button').at(1).props().onPress()
      expect(fakeCloseModal).toHaveBeenCalled()
    });
  });

});