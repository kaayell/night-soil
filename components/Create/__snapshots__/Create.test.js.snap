// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create should render a whole lot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#9AC0CD",
      "height": 400,
    }
  }
>
  <PoopRating
    onRatingChange={[Function]}
    selected={3}
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View>
      <FormLabel
        fontFamily="poppins-regular"
        labelStyle={
          Object {
            "color": "#FAFAFA",
            "fontSize": 15,
          }
        }
      >
        Duration In Minutes
      </FormLabel>
      <FormInput
        containerStyle={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#FAFAFA",
          }
        }
        inputStyle={
          Object {
            "color": "#FAFAFA",
            "paddingLeft": 10,
          }
        }
        keyboardType="numeric"
        normalizeFontSize={true}
        onChangeText={[Function]}
        placeholderTextColor="#FAFAFA"
        underlineColorAndroid="#FAFAFA"
      />
    </View>
    <View>
      <FormLabel
        fontFamily="poppins-regular"
        labelStyle={
          Object {
            "color": "#FAFAFA",
            "fontSize": 15,
          }
        }
      >
        Date
      </FormLabel>
      <DatePicker
        TouchableComponent={[Function]}
        allowFontScaling={true}
        androidMode="default"
        cancelBtnText="Cancel"
        confirmBtnText="Confirm"
        customStyles={
          Object {
            "dateInput": Object {
              "borderColor": "#FAFAFA",
              "borderLeftColor": "transparent",
              "borderRightColor": "transparent",
              "borderTopColor": "transparent",
              "borderWidth": 1,
            },
            "dateText": Object {
              "color": "#FAFAFA",
              "fontSize": 15,
            },
            "placeholderText": Object {
              "color": "#FAFAFA",
              "fontSize": 15,
            },
          }
        }
        date="08-17-2018"
        disabled={false}
        duration={300}
        format="MM-DD-YYYY"
        height={259}
        hideText={false}
        iconSource={
          Object {
            "testUri": "../../react-native-datepicker/date_icon.png",
          }
        }
        modalOnResponderTerminationRequest={[Function]}
        mode="date"
        onDateChange={[Function]}
        placeholder=""
        showIcon={false}
        style={
          Object {
            "borderColor": "#FAFAFA",
            "marginLeft": 15,
            "width": "91%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Button
      buttonStyle={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#FAFAFA",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginTop": 5,
        }
      }
      onPress={[Function]}
      textStyle={
        Object {
          "color": "#FAFAFA",
          "fontFamily": "poppins-medium",
          "fontSize": 15,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
      title="I'm at work"
    />
    <Button
      buttonStyle={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#FAFAFA",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginTop": 5,
        }
      }
      onPress={[Function]}
      textStyle={
        Object {
          "color": "#FAFAFA",
          "fontSize": 15,
          "fontWeight": "700",
        }
      }
      title="ADD"
    />
  </View>
</View>
`;
